packages/scene-composer/src/components/panels/scene-settings/MatterportIntegration.tsx

```diff
@@ -0,0 +1,220 @@
+      let sceneMetadata: Record<string, string> | undefined;
+      const getSceneresponse = await getSceneInfoFunction();
+      if (getSceneresponse) {
+        sceneCapabilities = getSceneresponse.capabilities;
+        sceneMetadata = getSceneresponse.sceneMetadata;
+      }
+
+      if (value === 'n/a') {
+        if (sceneCapabilities && sceneCapabilities.includes('MATTERPORT')) {
+          sceneCapabilities = sceneCapabilities.filter((capability) => capability !== 'MATTERPORT');
+        }
+
+        if (sceneMetadata && sceneMetadata.MATTERPORT_SECRET_ARN) {
+          delete sceneMetadata.MATTERPORT_SECRET_ARN;
+        }
+      } else {
+        if (sceneCapabilities) {
+          if (!sceneCapabilities.includes('MATTERPORT')) {
+            sceneCapabilities.push('MATTERPORT');
+          }
+        } else {
+          sceneCapabilities = ['MATTERPORT'];
+        }
+
+        if (sceneMetadata) {
+          sceneMetadata.MATTERPORT_SECRET_ARN = value;
+        } else {
+          sceneMetadata = { MATTERPORT_SECRET_ARN: value };
+        }
+      }
+
+      updateSceneFunction(sceneCapabilities, sceneMetadata);
+    }
+  }, []);
+
+  const selectedConnection = () => {
+    const selectedOption = connectionOptions.find((t) => t.value === selectedConnectionName);
+    if (selectedOption) {
+      return {
+        label: selectedOption.label,
+        value: selectedOption.value,
+      };
+    }
+    return null;
+  };
+
+  const onMatterportModelIdChange = useCallback(
+    (event) => {
+      const value = event.detail.value;
+      if (value !== matterportModelId) {
+        setMatterportModelId(value);
+        setDirty(true);
+      }
+    },
+    [setMatterportModelId, setDirty],
+  );
+
+  useEffect(() => {
+    if (dirty) {
+      setSceneProperty(KnownSceneProperty.MatterportModelId, matterportModelId !== '' ? matterportModelId : undefined);
+      setDirty(false);
+    }
+  }, [matterportModelId]);
+
+  return (
+    <React.Fragment>
+      <SpaceBetween size='s'>
+        <Box fontWeight='bold'>
+          {intl.formatMessage({ description: 'Sub-Section Header', defaultMessage: 'Matterport Integration' })}
+        </Box>
+        <Box>
+          {intl.formatMessage({
+            description: 'Sub-Section Description',
+            defaultMessage: 'Connecting your Matterport account enables viewing spaces in your TwinMaker scene. ',
+          })}
+        </Box>
+        {connectionOptions.length > 1 && (
+          <React.Fragment>
+            <FormField
+              label={intl.formatMessage({ description: 'Form Field label', defaultMessage: 'Connection Name' })}
+            >
+              <Select
+                selectedOption={selectedConnection()}
+                onChange={onConnectionNameChange}
+                options={connectionOptions}
+                selectedAriaLabel={intl.formatMessage({ defaultMessage: 'Selected', description: 'label' })}
+                disabled={connectionOptions.length === 0}
```

Comment:
with it's length ever be 0?

Reply:
Not actually unless there is some exception before populating the list of connections.